go/vt/vtgate/executor.go

```diff
@@ -241,8 +245,6 @@ func (e *Executor) handleExec(ctx context.Context, safeSession *SafeSession, sql
 		// TODO(sougou): change this flow to go through V3 functions
 		// which will allow us to benefit from the autocommitable flag.
 
-		queriesProcessed.Add([]string{"ShardDirect", "", ""}, 1)
-
 		if destKeyspace == "" {
```

Comment:
Minor nit -- this change loses the counter in the event that the function returns early due to an empty keyspace. Probably acceptable given the scope of the problem but just wanted to point it out.